### PR TITLE
Improved doc string for hyp2f1

### PR DESCRIPTION
**Pull request by ashwinpathak20**
I have changed the doc string of hyp2f1 function and included all the necessary details. Issue #6966

### CODE REVIEW

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -3035,6 +3035,52 @@ def add_newdoc(place, name, doc):
     hyp2f1(a, b, c, z)
 
     Gauss hypergeometric function 2F1(a, b; c; z).
```
Comment (person142) [inline scipy/special/add_newdocs.py]: Format here should be a short definition (that part is fine) followed by a longer definition. The series definition in the notes should be moved here to be the longer definition. See e.g. https://scipy.github.io/devdocs/generated/scipy.special.eval_jacobi.html#scipy.special.eval_jacobi.

File: scipy/special/add_newdocs.py [hunk 2]
```diff
@@ -3035,6 +3035,52 @@ def add_newdoc(place, name, doc):
...
+    b : double argument.
+
+    c : double argument.
+
+    z : double argument.
+
+    Returns
+    -------
+    y : double argument
+        y being the solution to the second-order linear ordinary differential equation.
```
Comment (person142) [inline scipy/special/add_newdocs.py]: You haven't told the user that this function satisfies a second-order linear equation, so this isn't a good way to go. Just say something like "values of the Gaussian hypergeometric function".

File: scipy/special/add_newdocs.py [hunk 3]
```diff
@@ -3035,6 +3035,52 @@ def add_newdoc(place, name, doc):
...
+
+    Notes
+    -----
+
+    .. math::
+
+        \mathrm{hyp2f1}(a, b, c, z) = 1 + \sum_{k=0}^inf. \frac {{{a+k}\choose{a}} {{b+k}}\choose{b}}}{{c+k}\choose{c}} z^(k+1).
```
Comment (person142) [inline scipy/special/add_newdocs.py]: Move to above, as noted in earlier comment.

File: scipy/special/add_newdocs.py [hunk 4]
```diff
@@ -3035,6 +3035,52 @@ def add_newdoc(place, name, doc):
...
+      Conditionally, a recurrence on c to make c-a-b > 0
+
+      z < -1  AMS 15.3.7 transformation applied (Travis Oliphant)
+         valid for b,a,c,(b-a) != integer and (c-a),(c-b) != negative integer
+
+      z >= 1 is rejected (unless special cases are present)
+    
+    See also
+    --------
+    hyp1f1 , hyp0f1.
```
Comment (person142) [inline scipy/special/add_newdocs.py]: Format needs to be different here; see `eval_jacobi` example linked above.

**Review by person142 — CHANGES_REQUESTED** [review]
Note that you need to run `generate_ufuncs.py` for these changes to actually show up in the published docs (as was mentioned in the mailing list, run it and commit the changes in a separate commit).

File: scipy/special/add_newdocs.py [hunk 5]
```diff
@@ -3035,6 +3035,52 @@ def add_newdoc(place, name, doc):
...
+        y being the solution to the second-order linear ordinary differential equation.
+
+    Notes
+    -----
+
+    .. math::
+
+        \mathrm{hyp2f1}(a, b, c, z) = 1 + \sum_{k=0}^inf. \frac {{{a+k}\choose{a}} {{b+k}}\choose{b}}}{{c+k}\choose{c}} z^(k+1).
+    
+    Cases addressed are
```
Comment (person142) [inline scipy/special/add_newdocs.py]: While I am pro discussing the algorithm in the notes, you need more description here for the user to be able to understand what's happening. In particular links to the DLMF (http://dlmf.nist.gov/) are very helpful. Also this appears to only discuss what happens for the real case.

File: scipy/signal/_peak_finding.py [hunk 6]
```diff
@@ -448,7 +448,7 @@ def find_peaks_cwt(vector, widths, wavelet=None, max_distances=None,
         If a relative maximum is not found within `max_distances`,
         there will be a gap. A ridge line is discontinued if there are more
         than `gap_thresh` points without connecting a new relative maximum.
-        Default is 2.
+        Default is widths[0].
```
Comment (ashwinpathak20) [inline scipy/signal/_peak_finding.py]: i am really sorry, i didn't realise that this was also included.

Comment (ashwinpathak20) [inline scipy/signal/_peak_finding.py]: Also, how do I know that I have run generate_ufuncs.py correctly?

File: scipy/special/add_newdocs.py [hunk 7]
```diff
@@ -3071,10 +3078,15 @@ def add_newdoc(place, name, doc):
...
-    
+
+      a,b,c or z can take up complex values too.
+      
+      http://dlmf.nist.gov/15.4
+
     See also
     --------
-    hyp1f1 , hyp0f1.
+    hyp0f1 : confluent hypergeometric function.
```
Comment (person142) [inline scipy/special/add_newdocs.py]: `hyp0f1` is the confluent hypergeometric *limit* function.

File: scipy/special/add_newdocs.py [hunk 8]
```diff
@@ -3071,10 +3078,15 @@ def add_newdoc(place, name, doc):
          valid for b,a,c,(b-a) != integer and (c-a),(c-b) != negative integer
 
       z >= 1 is rejected (unless special cases are present)
-    
+
+      a,b,c or z can take up complex values too.
```
Comment (person142) [inline scipy/special/add_newdocs.py]: `a`, `b`, and `c` cannot take complex values, only `z` can.

File: scipy/special/add_newdocs.py [hunk 9]
```diff
@@ -3036,28 +3036,35 @@ def add_newdoc(place, name, doc):
...
-    y : double argument
-        y being the solution to the second-order linear ordinary differential equation.
+    y : ndarray
+        y being the values of the gaussian hypergeometric function.
 
     Notes
     -----
 
-    .. math::
+    Gaussian function is defined on the disk |z|<1, and by analytic continuation elsewhere.
```
Comment (person142) [inline scipy/special/add_newdocs.py]: The Gaussian function is `e^(-x**2)`. Should be Gaussian hypergeometric function.

File: scipy/special/add_newdocs.py [hunk 10]
```diff
@@ -3036,28 +3036,35 @@ def add_newdoc(place, name, doc):
...
-    z : double argument.
+    z : array_like
+        Parameter
 
     Returns
     -------
-    y : double argument
-        y being the solution to the second-order linear ordinary differential equation.
+    y : ndarray
+        y being the values of the gaussian hypergeometric function.
```
Comment (person142) [inline scipy/special/add_newdocs.py]: Just "values of the Gaussian hypergeometric function"

File: scipy/special/add_newdocs.py (hunk 8 above)
Comment (person142) [inline scipy/special/add_newdocs.py]: Also what can or cannot take complex values should be specified in the parameters section.

File: scipy/special/add_newdocs.py [hunk 11]
```diff
@@ -3036,28 +3036,35 @@ def add_newdoc(place, name, doc):
 
     Gauss hypergeometric function 2F1(a, b; c; z).
 
+    \mathrm{hyp2f1}(a, b, c, z) = 1 + \sum_{k=0}^inf. \frac {{{a+k}\choose{a}} {{b+k}}\choose{b}}}{{c+k}\choose{c}} z^(k+1).
```
Comment (person142) [inline scipy/special/add_newdocs.py]: Make this a complete sentence. Something like "The Gaussian hypergeometric function is defined as...". Also I wouldn't use binomial coefficients here, the Pochhammer symbol is more appropriate.

File: scipy/special/add_newdocs.py [hunk 12]
```diff
@@ -3036,28 +3036,35 @@ def add_newdoc(place, name, doc):
...
-    b : double argument.
+    b : array_like
+        Parameter
 
-    c : double argument.
+    c : array_like
+        Parameter
 
-    z : double argument.
+    z : array_like
```
Comment (person142) [inline scipy/special/add_newdocs.py]: "real or complex array_like", similar for the others.

File: scipy/special/add_newdocs.py [hunk 13]
```diff
@@ -3036,28 +3036,35 @@ def add_newdoc(place, name, doc):
...
+        Parameter
```
Comment (person142) [inline scipy/special/add_newdocs.py]: This is the argument, not a parameter.

File: scipy/special/add_newdocs.py (hunk 11 above)
Comment (person142) [inline scipy/special/add_newdocs.py]: Also you want to mention the analytic continuation and branch cut here as they are part of the definition of the function.

**Comment by person142** [comment]
Changes in `_ufuncs_cxx.pxd ` etc. shouldn't be here. Guessing it's because of the merge commits; instead of merging in the master branch you should rebase on it.

**Comment by ashwinpathak20** [comment]
@person142  I am not able to understand why my code is not able to pass the checks , can you please help me out here?

**Comment by ev-br** [comment]
You can click through the CI status to see the test log: *Docstring contains a non-printable character '\x0c'! Maybe forgot r"""?*

**Comment by ashwinpathak20** [comment]
I did include r at the starting , after defining the function in add_newdocs.py but still the same result.

**Comment by person142** [comment]
There are at least a couple more errors, e.g. not all of the math is in math mode.

**Comment by ashwinpathak20** [comment]
Is that the reason of not passing the checks or the math code is incorrect according to the hyp2f1?

**Comment by person142** [comment]
Let's try to get this sorted out. The erroneous +13,737 LOC and large number of commits is making review a bit of a pain, so we'll start there. Some things for the future:

- It looks like you did your development on master; it's better to do it on a topic branch.
- Don't merge the master branch to resolve conflicts--instead rebase on master (`git rebase`).

Probably the easiest way to fix this now is to:

- Rename this branch to something else (`git branch -m <oldbranch> <newbranch>`). Let's call it `hyp2f1-doc` for concreteness.
- Grab a new copy of the master branch from the main SciPy repo so that you have a clean master again.
- Rebase `hyp2f1-doc` on the clean master with `git rebase -i master hyp2f1-doc`. Drop every commit that isn't modifying `add_newdocs.py`, and squash all of the commits modifying `add_newdocs.py`. Also modify the commit message to be formatted according to https://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html#writing-the-commit-message.
- There should now be one commit that only modifies `add_newdocs.py`. Go run `generate_ufuncs.py` and commit the changes with the message `GEN: run generate_ufuncs.py`.
- Push the changes, force pushing (`git push -f`) is fine.

**Comment by ashwinpathak20** [comment]
When i did rebasing of the branch , in the interactive mode, there was noop present rather then the commits, so I was not able to squash the commits.

**Comment by person142** [comment]
Please don't push more changes until this is sorted out; it is making the situation worse.

> When i did rebasing of the branch , in the interactive mode, there was noop present rather then the commits

I am not sure what you mean here. When you run `git rebase -i` it opens a text editor which shows the commits you have added since your topic branch diverged from master. Did you create a new topic branch as instructed?

**Comment by ashwinpathak20** [comment]
yes i followed the same steps.... can i make a new pull request instead?

**Comment by person142** [comment]
> can i make a new pull request instead?

Sure. Just be careful of the points mentioned above.

**Comment by ashwinpathak20** [comment]
I am really sorry for the last commits , please ignore them

**Pull request closed without merging**